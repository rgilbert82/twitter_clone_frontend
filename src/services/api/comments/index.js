export { default as getCommentsAPI } from './getComments.js';
export { default as createCommentAPI } from './createComment.js';
export { default as editCommentAPI } from './editComment.js';
export { default as deleteCommentAPI } from './deleteComment.js';
