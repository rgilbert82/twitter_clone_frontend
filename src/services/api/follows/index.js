export { default as getFollowAPI } from './getFollow.js';
export { default as createFollowAPI } from './createFollow.js';
export { default as deleteFollowAPI } from './deleteFollow.js';
export { default as getUserFollowersAPI } from './getUserFollowers.js';
export { default as getUserFollowingsAPI } from './getUserFollowings.js';
