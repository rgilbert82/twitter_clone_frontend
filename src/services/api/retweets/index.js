export { default as createRetweetAPI } from './createRetweet.js';
export { default as deleteRetweetAPI } from './deleteRetweet.js';
