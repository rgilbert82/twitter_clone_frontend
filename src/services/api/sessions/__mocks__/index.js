export { default as loginAPI } from './login.js';
export { default as logoutAPI } from './logout.js';
export { default as getCurrentUserAPI } from './getCurrentUser.js';
