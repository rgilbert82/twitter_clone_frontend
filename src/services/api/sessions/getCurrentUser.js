import { get_API_URL } from '../../misc';

export default () => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/sessions/1`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(JSON.parse(request.responseText));
        } else {
          reject(request.responseText);
        }
      }
    };

    request.open('GET', path);
    request.withCredentials = true;
    request.setRequestHeader('Authorization', document.cookie || '');
    request.send();
  });
}
