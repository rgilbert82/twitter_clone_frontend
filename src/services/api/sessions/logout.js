import { get_API_URL } from '../../misc';

export default () => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/sessions/1`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(request.responseText);
        } else {
          reject(request.responseText);
        }
      }
    };

    request.open('DELETE', path);
    request.withCredentials = true;
    request.setRequestHeader('Authorization', document.cookie || '');
    request.send();
  });
}
