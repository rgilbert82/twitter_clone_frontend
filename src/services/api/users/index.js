export { default as getUsersAPI } from './getUsers.js';
export { default as getUserAPI } from './getUser.js';
export { default as createUserAPI } from './createUser.js';
export { default as updateUserAPI } from './updateUser.js';
export { default as getUserTweetsAPI } from './getUserTweets.js';
export { default as deleteUserAPI } from './deleteUser.js';
