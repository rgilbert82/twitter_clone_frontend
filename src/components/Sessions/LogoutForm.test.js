jest.mock('../../services/api/sessions');

import React from 'react';
import { LogoutForm } from '.';
import { shallow } from 'enzyme';
import { logoutAPI } from '../../services/api/sessions';

describe(LogoutForm, () => {
  let appLogoutMock = jest.fn();
  let logoutSpy = jest.spyOn(LogoutForm.prototype, 'logout');
  let component;

  beforeEach(() => {
    component = shallow(<LogoutForm appLogout={appLogoutMock} />);
  });

  afterAll(() => {
    logoutSpy.mockClear();
  });

  it('renders properly', () => {
    expect(component).toMatchSnapshot();
  });

  it('calls logout on button click', () => {
    component.find('#submit_logout_form').simulate('click');
    expect(logoutSpy).toHaveBeenCalled();
  });

  it('logs out the user', () => {
    expect.assertions(1);
    component.instance().logout()
      .then(() => {
        expect(appLogoutMock).toBeCalled();
      });
  });
});
