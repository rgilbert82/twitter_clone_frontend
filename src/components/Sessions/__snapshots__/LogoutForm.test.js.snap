// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class LogoutForm extends _react2.default.Component {
  constructor(props) {
    super(props);
    this.logout = this.logout.bind(this);
  }

  logout() {
    return (0, _sessions.logoutAPI)().
    then(() => {
      this.props.appLogout();
    }).catch(() => {
      this.props.appLogout();
    });
  }

  render() {
    return (
      _react2.default.createElement(_reactBootstrap.Navbar.Form, { id: 'logout_form', __source: { fileName: _jsxFileName, lineNumber: 22 }, __self: this },
        _react2.default.createElement(_reactBootstrap.Button, { id: 'submit_logout_form', onClick: this.logout, __source: { fileName: _jsxFileName, lineNumber: 23 }, __self: this }, 'Logout')));


  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogoutForm
    appLogout={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        id="submit_logout_form"
        onClick={[Function]}
>
        Logout
</Button>,
      "componentClass": "div",
      "id": "logout_form",
      "pullLeft": false,
      "pullRight": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": false,
        "block": false,
        "bsClass": "btn",
        "bsStyle": "default",
        "children": "Logout",
        "disabled": false,
        "id": "submit_logout_form",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Logout",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          id="submit_logout_form"
          onClick={[Function]}
>
          Logout
</Button>,
        "componentClass": "div",
        "id": "logout_form",
        "pullLeft": false,
        "pullRight": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Logout",
          "disabled": false,
          "id": "submit_logout_form",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Logout",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
